app/components/user-selector.js

```diff
@@ -7,27 +7,48 @@ export default class UserSelectorComponent extends Component {
   // Track the currently selected user's ID
   @tracked selectedUserId;
 
-  // Initialize the component with the appropriate selected user ID
   constructor() {
     super(...arguments);
-    // Set the selected user ID based on the provided task
-    // Checking if this exists - this is where my test breaks down - as its not getting args
-    this.selectedUserId = this.args.task.user.id
-      ? this.args.task.user.id
-      : null;
+    // Moved initialization logic into a separate method for clarity
+    this.initializeSelectedUserId();
+  }
+
+  /**
+   * Initialize the selected user ID based on the provided task argument.
+   * This method ensures the component starts with the correct user selection,
+   * handling potential null or undefined values gracefully.
+   */
+  initializeSelectedUserId() {
+    // Set the selected user ID to the user's ID if available, or null if unassigned
+    this.selectedUserId = this.args.task?.user?.id || null;
   }
 
   @action
   handleChange(event) {
-    // Update the selected user ID when the selection changes - store it locally until we grab it in edit-task controller
-    this.selectedUserId = event.target.value;
-
+    // Convert the selected value to null if it represents unassignment, otherwise remain with same value
+    this.selectedUserId =
+      event.target.value === 'null' ? null : event.target.value;
   }
 
+  /**
+   * Returns the list of options for the user selector dropdown.
+   * Includes a Base option to show either "Unassigned" or "Set as Unassigned" if the associated user id is null
+   * If there is a selectedUserId (user already associated) - the baseOption will be "Set as Unassigned"
+   * If there is no id - the baseOption will be "Unassigned"
+   */
   get userOptions() {
-    // I want to show the option to set as unassigned when a task is already assigned else show unassigned
-    return this.selectedUserId
-      ? [{ id: null, name: 'Set as Unassigned' }, ...this.args.users]
-      : [{ id: null, name: 'Unassigned' }, ...this.args.users];
+    const baseOption = {
+      id: null,
+      name: this.selectedUserId ? 'Set as Unassigned' : 'Unassigned',
+    };
+
+    // Map over the real users provided in arguments to format them for the dropdown
+    const userOptions = this.args.users.map((user) => ({
+      id: user.id, // Use user ID as the value
+      name: `${user.firstName} ${user.lastName}`,
```

Comment:
if `user` comes as a model, we can use `name` from getter model.

Reply:
I never actually added a name getter - will add one now and change that up (what you reference above - thanks for the tipp) 👍🏼